c10/core/TensorImpl.h

```diff
@@ -558,6 +559,11 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
       ;
 #endif
 
+  TENSORIMPL_MAYBE_VIRTUAL inline c10::SymIntArrayRef sym_sizes() const {
+    return c10::SymIntArrayRef(
+        reinterpret_cast<const c10::SymInt*>(sizes().data()), sizes().size());
```

Comment:
Given the most recent discussion by @Chillee and @zou3519, we shouldn't actually compute `sym_sizes` in this way; instead, we should generate IR nodes that correspond to `(x.shape[0], x.shape[1], ...)` which is sufficiently symbolic. I guess it might be slightly better if we only did this once per tensor but it's a bit annoying to do so we'll have to rely on CSE.

Reply:
we will! We didn't have IR nodes when I started this PR. I just recently landed Ansley's PR with Size IR